internal/provider/resource_auth0_resource_server.go

```diff
@@ -147,53 +151,42 @@ func createResourceServer(ctx context.Context, d *schema.ResourceData, m interfa
 
 func readResourceServer(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	api := m.(*management.Management)
+
 	resourceServer, err := api.ResourceServer.Read(d.Id())
 	if err != nil {
-		if mErr, ok := err.(management.Error); ok {
-			if mErr.Status() == http.StatusNotFound {
-				d.SetId("")
-				return nil
-			}
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			d.SetId("")
+			return nil
 		}
 		return diag.FromErr(err)
 	}
 
 	result := multierror.Append(
-		d.Set("name", resourceServer.Name),
-		d.Set("identifier", resourceServer.Identifier),
-		d.Set("scopes", func() []map[string]interface{} {
-			scopes := make([]map[string]interface{}, len(resourceServer.Scopes))
-			for index, scope := range resourceServer.Scopes {
-				scopes[index] = map[string]interface{}{
-					"value":       scope.Value,
-					"description": scope.Description,
-				}
-			}
-			return scopes
-		}()),
-		d.Set("signing_alg", resourceServer.SigningAlgorithm),
-		d.Set("signing_secret", resourceServer.SigningSecret),
-		d.Set("allow_offline_access", resourceServer.AllowOfflineAccess),
-		d.Set("token_lifetime", resourceServer.TokenLifetime),
-		d.Set("token_lifetime_for_web", resourceServer.TokenLifetimeForWeb),
+		d.Set("name", resourceServer.GetName()),
```

Comment:
Changed these to use the getters for extra safety in case we're trying to access a nil pointer.